[Pages] Custom domain debugging (#8306)
cloudflare/cloudflare-docs: content/pages/platform/debugging-pages.md

### Things blocking HTTP validation

Pages uses HTTP validation which means we need to hit a HTTP endpoint, if something is in the way (such as Access, a redirect, a Worker, etc.) then it means the validation cannot be completed.

To check this, simply run a curl against your domain hitting `/.well-known/acme-challenge/randomstring`, example:
```sh
$ curl -I https://example.com/.well-known/acme-challenge/randomstring

HTTP/2 302
date: Mon, 03 Apr 2023 08:37:39 GMT
location: https://example.cloudflareaccess.com/cdn-cgi/access/login/example.com?kid=...&redirect_url=%2F.well-known%2Facme-challenge%2F...
access-control-allow-credentials: true
cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
server: cloudflare
cf-ray: 7b1ffdaa8ad60693-MAN
```

As we can see in the example above, we're redirecting to Cloudflare Access (as shown by the `Location` header). In this case, we will need to disable Access over the domain until it's verified. Once it's verified, Access can be re-applied above.

> You will need to do the same kind of thing for Redirect Rules or a Worker example too.

Reviewer: ```suggestion
You will need to follow the same instructions for redirects or a Worker example.
```

### Missing CAA records

> If nothing is blocking the HTTP validation, then you may be missing CAA records. If you have disabled [Universal SSL](/ssl/edge-certificates/universal-ssl/) or use an external provider then this is likely.

Reviewer: ```suggestion
If nothing is blocking the HTTP validation, then you may be missing Certification Authority Authorization (CAA) records. If you have disabled [Universal SSL](/ssl/edge-certificates/universal-ssl/) or use an external provider then this is likely.
```

> To check, we can do a `dig` on the custom domain's apex (or zone if this is a liberated zone). For example:

Author: preventing overlap of line 112
```suggestion
To check this, run a `dig` on the custom domain's apex (or zone, if this is a liberated zone). For example:
```

```
$ dig example.com

dig CAA example.com

; <<>> DiG 9.10.6 <<>> CAA example.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 59018
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;example.com.		IN	CAA

;; ANSWER SECTION:
example.com.	300	IN	CAA	0 issue "amazon.com"

;; Query time: 92 msec
;; SERVER: 127.0.2.2#53(127.0.2.2)
;; WHEN: Mon Apr 03 10:15:51 BST 2023
;; MSG SIZE  rcvd: 76
```

> As we can see in this example, there's only a single CAA record which is allowing Amazon to issue ceritficates.

Reviewer: ```suggestion
In the above example, there is only a single CAA record which is allowing Amazon to issue ceritficates.
```

> To resolve this, you will need to add the following CAA records which allows all of the CAs we use to issue ceritificates:

Reviewer: ```suggestion
To resolve this, you will need to add the following CAA records which allows all of the Certificate Authorities (CAs) Cloudflare uses to issue ceritificates:
```

```
example.com.            300     IN      CAA     0 issue "comodoca.com"
example.com.            300     IN      CAA     0 issue "digicert.com; cansignhttpexchanges=yes"
example.com.            300     IN      CAA     0 issue "letsencrypt.org"
example.com.            300     IN      CAA     0 issue "pki.goog; cansignhttpexchanges=yes"
example.com.            300     IN      CAA     0 issuewild "comodoca.com"
example.com.            300     IN      CAA     0 issuewild "digicert.com; cansignhttpexchanges=yes"
example.com.            300     IN      CAA     0 issuewild "letsencrypt.org"
example.com.            300     IN      CAA     0 issuewild "pki.goog; cansignhttpexchanges=yes"
```

> ### Still having trouble?
> 
> If you've done the steps above and your domain is still stuck pending after 15 minutes, please contact support through the [Support Portal](https://dash.cloudflare.com/?to=/:account/support).

Reviewer: ```suggestion
{{<Aside type="warning" header="Still having issues">}}
If you have done the steps above and your domain is still verifying after 15 minutes, contact support through the [Support Portal](https://dash.cloudflare.com/?to=/:account/support).
{{</Aside>}}
```

Author: We should also mention Discord here (since I'm there and I'll find it in like 10 seconds)

Altered suggestion:
```suggestion
{{<Aside type="warning" header="Still having issues">}}

If you have done the steps above and your domain is still verifying after 15 minutes, join our [Discord](https://discord.cloudflare.com) or contact support through the [Support Portal](https://dash.cloudflare.com/?to=/:account/support).

{{</Aside>}}
```
